examples/model_selection/plot_precision_recall.py

```diff
@@ -111,44 +114,89 @@
                                  random_state=random_state))
 y_score = classifier.fit(X_train, y_train).decision_function(X_test)
 
-# Compute Precision-Recall and plot curve
+
+def reversed_precision_recall_curve(y_true, y_score):
+    """Helper function to return precision, recall and thresholds
+    in reverse order"""
+    p, r, t = precision_recall_curve(y_true, y_score)
+    return p[::-1], r[::-1], t[::-1]
+
+
+def get_circle_coords(p, r):
+    """Get coordinates of operating points chosen for 11-point interpolated
+    average precision"""
+    recall_circles = list()
+    precision_circles = list()
+    for threshold in np.arange(0, 1.1, 0.1):
+        i = sum(r[1:] >= threshold)
+        recall_circles.append(r[-i])
+        precision_circles.append(p[-i])
+    return recall_circles, precision_circles
+
+
+def fill_beneath_step(x, y, color, alpha=0.2):
+    """Fill an area underneath a step function"""
+    x_long = [v for v in x for _ in (0, 1)][:-1]
+    y_long = [v for v in y for _ in (0, 1)][1:]
+    plt.fill_between(x_long, 0, y_long, alpha=alpha, color=color)
+
+
+# Compute Precision-Recall, average precision and eleven-point interpolated
+# average precision
 precision = dict()
 recall = dict()
 average_precision = dict()
+interpolated_average_precision = dict()
 for i in range(n_classes):
-    precision[i], recall[i], _ = precision_recall_curve(y_test[:, i],
+    precision[i], recall[i], _ = reversed_precision_recall_curve(y_test[:, i],
```

Comment:
I don't understand the need for reversing. It's a pity, because it makes the example harder to understand. That example is already getting a bit challenging: not an ideal introduction to precision-recall.